src/components/ButtonGroupPicker/__test__/buttonGroupPicker.spec.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+import { Component as ButtonGroupPicker } from '../index';
+import ButtonOption from '../../ButtonOption/index';
+
+describe('<ButtonGroupPicker />', () => {
+    it('should render a label when label prop is passed', () => {
+        const component = mount(<ButtonGroupPicker label="My label" />);
+        expect(component.find('legend').length).toBe(1);
+    });
+
+    it('should render bottom help text when bottomHelpText prop is passed', () => {
+        const component = mount(<ButtonGroupPicker bottomHelpText="My help text" />);
+        expect(component.find('div[children="My help text"]').length).toBe(1);
+    });
+
+    it('should render required asterisk when required prop is true', () => {
+        const component = mount(<ButtonGroupPicker label="My label" required />);
+        expect(component.find('abbr[title="required"]').length).toBe(1);
+    });
+
+    it('should pass type checkbox in context when multiple is true', () => {
+        const component = shallow(<ButtonGroupPicker multiple />);
+        const context = component.instance().getContext();
+        expect(context.type).toBe('checkbox');
+    });
+
+    it('should pass type radio in context when multiple is false', () => {
+        const component = shallow(<ButtonGroupPicker />);
+        const context = component.instance().getContext();
+        expect(context.type).toBe('radio');
+    });
+
+    it('should render an error text when error prop is passed', () => {
+        const component = mount(<ButtonGroupPicker error="error" />);
+        expect(component.find('div[children="error"]').length).toBe(1);
+    });
+
+    it('should set the error message id as id for the error text', () => {
+        const component = mount(<ButtonGroupPicker error="error" />);
+        const errorMessageId = component.instance().errorMessageId;
```

Comment:
same ^^ let's try to avoid treatment with instance variables (implementation details that could change for example if we migrate this component to functional) and instead mock `uniqueId` function and check that the prop id is passed to StyledErrorText is what the mock returned